bin/terminal.js

```diff
@@ -36,21 +36,53 @@ function listPorts() {
   });
 };
 
-function createPort() {
-  if (!args.port) {
-    args.outputHelp();
-    args.missingArgument('port');
-    process.exit(-1);
+function setupPort() {
+  if(args.port) {
+    createPort(args.port);
   }
 
+  SerialPort.list((err, ports) => {
+    if (err) {
+      console.error('Error listing ports, and missing port argument.', err);
+      args.outputHelp();
+      args.missingArgument('port');
+      process.exit(-1);
+    } else {
+      if(ports.length > 0) {
+        var portSelection = new List({
+          name: 'serial-port-selection',
+          message: 'Select a serial port to open',
+          choices: ports.map((port, i) => `[${i + 1}]\t${port.comName}\t${port.pnpId || ''}\t${port.manufacturer || ''}`)
+        });
+  
+        portSelection.run()
+          .then(answer => {
+            const choice = answer.split('\t')[1];
+            console.log(`Opening serial port: ${choice}`);
+            createPort(choice);
+          })
+          .catch(error => {
+            console.log(`Could not select a port: ${error}`);
+            process.exit(-2);
```

Comment:
Not sure what your policy is on the error code numbering that should be returned. Could use feedback.